reframe/core/schedulers/pbs.py

```diff
@@ -209,6 +211,19 @@ def output_ready(job):
                 if job.cancelled or output_ready(job):
                     self.log(f'Assuming job {job.jobid} completed')
                     job._completed = True
+                if is_pbs_pro:
+                    # With PBS Pro we can obtain the exit status of the job,
+                    # in case it actually failed.
+                    extended_info = osext.run_command(
+                        f'qstat -xf {job.jobid}'
+                    )
+                    exit_status_match = re.search(
+                        r'^ *Exit_status *= *(?P<exit_status>\d+)',
+                        extended_info.stdout,
+                        flags=re.MULTILINE,
+                    )
+                    if exit_status_match:
+                        job._exitcode = int(exit_status_match.group('exit_status'))
```

Comment:
I would not use the `if` here but rather let inheritance take care of it.

```python
job._exitcode = self._query_exit_code(job.jobid)
```

Where each of the PBS-based backends implements `_query_exit_code()` accordingly. The `torque` backend will simply return `None`.